### PR TITLE
[es-shims] Add support for the `String.prototype` Annex B methods

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**

### CODE REVIEW

File: packages/babel-plugin-polyfill-es-shims/data/polyfills.json [hunk 1]
```diff
@@ -824,5 +823,174 @@
...
+    "firefox": "17",
+    "safari": "6",
+    "node": "0.10",
+    "android": "4",
+    "ios": "7",
+    "phantom": "2",
+    "samsung": "1",
+    "electron": "0.20"
+  },
+  "String.prototype.sup": {
```
Comment (ljharb) [inline packages/babel-plugin-polyfill-es-shims/data/polyfills.json:983]: I believe all of these work in ie 10 and 11 (but haven’t checked lately)

**Review by ljharb — APPROVED** [review]

File: packages/babel-plugin-polyfill-es-shims/data/polyfills.json (hunk 1 above)
Comment (nicolo-ribaudo) [reply packages/babel-plugin-polyfill-es-shims/data/polyfills.json:983]: I don't have access to IE to test it, but this data is generated from `compat-table`.

Comment (ljharb) [reply packages/babel-plugin-polyfill-es-shims/data/polyfills.json:983]: Oops, I’ll update the compat table then, please hold :-)

Comment (ljharb) [reply packages/babel-plugin-polyfill-es-shims/data/polyfills.json:983]: The data in the compat table under "name: 'String.prototype HTML methods'" includes `ie10: true,` for one test, and `ie11: false,` for another two which means all IE versions require the polyfill. What am i missing?

Comment (nicolo-ribaudo) [reply packages/babel-plugin-polyfill-es-shims/data/polyfills.json:983]: It might be a bug in the script which extracts the data, let me check :thinking:

Comment (nicolo-ribaudo) [reply packages/babel-plugin-polyfill-es-shims/data/polyfills.json:983]: Ahhh, this file contains _supported_ versions. IE is not marked as supported, so it always include the polyfill.

Comment (nicolo-ribaudo) [reply packages/babel-plugin-polyfill-es-shims/data/polyfills.json:984]: I added a test

Comment (ljharb) [reply packages/babel-plugin-polyfill-es-shims/data/polyfills.json:984]: ahhh ok thanks, that wasn't clear to me

**Pull request merged**